File: server/server/src/main/kotlin/org/jetbrains/bazel/server/bsp/info/BspInfo.java
package org.jetbrains.bazel.server.bsp.info;

import java.nio.file.Path;

public class BspInfo {

  private final Path bspProjectRoot;

  public BspInfo(Path bspProjectRoot) {
    this.bspProjectRoot = bspProjectRoot;
  }

  public Path bspProjectRoot() {
    return bspProjectRoot;
  }

  public Path bazelBspDir() {
    return bspProjectRoot().resolve(".bazelbsp");
  }
}

File: server/server/src/main/kotlin/org/jetbrains/bazel/server/bsp/info/BspInfo.kt
package org.jetbrains.bazel.server.bsp.info

import org.jetbrains.annotations.ApiStatus
import org.jetbrains.bazel.commons.constants.Constants
import java.nio.file.Path

@ApiStatus.Internal
open class BspInfo(val bspProjectRoot: Path) {
  open val bazelBspDir: Path = bspProjectRoot.resolve(Constants.DOT_BAZELBSP_DIR_NAME)
}
